### PR TITLE
Fix agent keep executing unknown command

### PR DESCRIPTION
**Pull request by john-the-dev**
### Background

https://github.com/Significant-Gravitas/Auto-GPT/issues/4886

### Changes

Check if the command result indicates the command does not exist. If it does, suggest in result not to run it again.

### Documentation

Simple change. No need for extra documentation.

### Test Plan

The steps to reproduce the error is not fully reliable due to unreliability of LLM, but what I did: create an agent called ColoringPageGPT, goals: create a Python script which draws a coloring page; run the script to create a coloring page. The agent will then try to run a command "draw_coloring_page" which does not exist but it will keep trying. The fix is to have a message to ask it not to keep trying as the command obviously does not exist.

### PR Quality Checklist
- [Y] My pull request is atomic and focuses on a single change.
- [Y] I have thoroughly tested my changes with multiple different prompts.
- [Y] I have considered potential risks and mitigations for my changes.
- [Y] I have documented my changes clearly and comprehensively.
- [Y] I have not snuck in any "extra" small tweaks changes. 
- [Y] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

No unit tests found for the function to be modified: `start_interaction_loop`. Performed manual tests.

### CODE REVIEW

File: autogpt/agent/agent.py [hunk 1]
```diff
@@ -284,7 +284,10 @@ def signal_handler(signum, frame):
                 memory_tlength = count_string_tokens(
                     str(self.history.summary_message()), self.config.smart_llm
                 )
-                if result_tlength + memory_tlength + 600 > self.smart_token_limit:
+                if str(command_result).startswith("Unknown command"):
```
Comment (Pwuts) [inline autogpt/agent/agent.py:287]: a search for `Unknown command` returned 0 results in the codebase, so there is no way this will work

**Comment by Pwuts** [comment]
We already do this:
https://github.com/Significant-Gravitas/Auto-GPT/blob/a0f5aa942de3a94ce6a173c784d7a73b17a134a6/autogpt/agents/agent.py#L393-L396

File: autogpt/agent/agent.py (hunk 1 above)
Comment (john-the-dev) [reply autogpt/agent/agent.py:287]: It was here: https://github.com/Significant-Gravitas/Auto-GPT/blob/8bce02736b7fcc3ef14809fa935c9f147ccc2e0a/autogpt/app.py#L131 but it seems the issue was fixed with this PR: https://github.com/Significant-Gravitas/Auto-GPT/commit/4177c37b51d2f2c8f19c929cadd2609262682605#diff-2c87cd30823d1d2b44a5130d7d1a788b97928a00ff4dc84e43fbfb0e2f2c1559R394. All good then. Thanks, Pwuts.

**Comment by john-the-dev** [comment]
> We already do this:
> 
> https://github.com/Significant-Gravitas/Auto-GPT/blob/a0f5aa942de3a94ce6a173c784d7a73b17a134a6/autogpt/agents/agent.py#L393-L396

Yes, saw it. Thanks.

**Pull request closed without merging**